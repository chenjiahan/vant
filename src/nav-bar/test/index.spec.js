import NavBar from '..';
import { mount, mockGetBoundingClientRect } from '../../../test';

test('should allow render left/right slot', () => {
  const wrapper = mount(NavBar, {
    scopedSlots: {
      left: () => 'Custom Left',
      right: () => 'Custom Right',
    },
  });

  expect(wrapper).toMatchSnapshot();
});

test('should allow render title slot', () => {
  const wrapper = mount(NavBar, {
    scopedSlots: {
      title: () => 'Custom Title',
    },
  });

  expect(wrapper).toMatchSnapshot();
});

test('should render placeholder element when enabling placeholder prop', () => {
  const restore = mockGetBoundingClientRect({ height: 50 });

  const wrapper = mount(NavBar, {
    propsData: {
      fixed: true,
      placeholder: true,
    },
  });

  expect(wrapper).toMatchSnapshot();

  restore();
});

test('should emit click-left event when clicking left text', () => {
  const wrapper = mount(NavBar, {
    propsData: {
      leftText: 'left',
    },
  });

  wrapper.find('.van-nav-bar__left').trigger('click');
  expect(wrapper.emitted('click-left')).toBeTruthy();
});

test('should emit click-right event when clicking right text', () => {
  const wrapper = mount(NavBar, {
    propsData: {
      rightText: 'right',
    },
  });

  wrapper.find('.van-nav-bar__right').trigger('click');
  expect(wrapper.emitted('click-right')).toBeTruthy();
});

test('should add safe-area-inset-top classname when using safe-area-inset-top prop', () => {
  const wrapper = mount(NavBar, {
    propsData: {
      safeAreaInsetTop: true,
    },
  });

  expect(wrapper.contains('.van-nav-bar--safe-area-inset-top')).toBeTruthy();
});
