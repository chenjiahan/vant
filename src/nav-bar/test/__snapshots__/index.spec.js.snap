// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should allow render left/right slot 1`] = `
<div class="van-nav-bar van-hairline--bottom">
  <div class="van-nav-bar__content">
    <div class="van-nav-bar__left">Custom Left</div>
    <div class="van-nav-bar__title van-ellipsis"></div>
    <div class="van-nav-bar__right">Custom Right</div>
  </div>
</div>
`;

exports[`should allow render title slot 1`] = `
<div class="van-nav-bar van-hairline--bottom">
  <div class="van-nav-bar__content">
    <div class="van-nav-bar__title van-ellipsis">Custom Title</div>
  </div>
</div>
`;

exports[`should render placeholder element when enabling placeholder prop 1`] = `
<div class="van-nav-bar__placeholder" style="height: 50px;">
  <div class="van-nav-bar van-nav-bar--fixed van-hairline--bottom">
    <div class="van-nav-bar__content">
      <div class="van-nav-bar__title van-ellipsis"></div>
    </div>
  </div>
</div>
`;
